main/forum/editthread.php

```diff
@@ -0,0 +1,220 @@
+}
+// 4. anonymous posts are not allowed and the user is not logged in
+if (!$_user['user_id'] AND $current_forum['allow_anonymous'] <> 1) {
+    api_not_allowed();
+}
+
+// 5. Check user access
+if ($current_forum['forum_of_group'] != 0) {
+    $show_forum = GroupManager::user_has_access(
+        api_get_user_id(),
+        $current_forum['forum_of_group'],
+        GroupManager::GROUP_TOOL_FORUM
+    );
+    if (!$show_forum) {
+        api_not_allowed();
+    }
+}
+
+// 6. Invited users can't create new threads
+if (api_is_invitee()) {
+    api_not_allowed(true);
+}
+
+$groupId = api_get_group_id();
+if (!empty($groupId)) {
+    $group_properties = GroupManager :: get_group_properties($groupId);
+    $interbreadcrumb[] = array('url' => '../group/group.php?'.api_get_cidreq(), 'name' => get_lang('Groups'));
+    $interbreadcrumb[] = array('url' => '../group/group_space.php?'.api_get_cidreq(), 'name' => get_lang('GroupSpace').' '.$group_properties['name']);
+    $interbreadcrumb[] = array('url' => 'viewforum.php?'.api_get_cidreq().'&forum='.Security::remove_XSS($_GET['forum']), 'name' => $current_forum['forum_title']);
+    $interbreadcrumb[] = array('url' => 'newthread.php?'.api_get_cidreq().'&forum='.Security::remove_XSS($_GET['forum']),'name' => get_lang('EditThread'));
+} else {
+    $interbreadcrumb[] = array('url' => 'index.php?'.api_get_cidreq(), 'name' => $nameTools);
+    $interbreadcrumb[] = array('url' => 'viewforumcategory.php?'.api_get_cidreq().'&forumcategory='.$current_forum_category['cat_id'], 'name' => $current_forum_category['cat_title']);
+    $interbreadcrumb[] = array('url' => 'viewforum.php?'.api_get_cidreq().'&forum='.Security::remove_XSS($_GET['forum']), 'name' => $current_forum['forum_title']);
+    $interbreadcrumb[] = array('url' => '#', 'name' => get_lang('EditThread'));
+}
+
+/* Resource Linker */
+
+if (isset($_POST['add_resources']) AND $_POST['add_resources'] == get_lang('Resources')) {
+    $_SESSION['formelements']	= $_POST;
+    $_SESSION['origin']			= $_SERVER['REQUEST_URI'];
+    $_SESSION['breadcrumbs']	= $interbreadcrumb;
+    header('Location: ../resourcelinker/resourcelinker.php');
+    exit;
+}
+
+/* Header */
+
+$htmlHeadXtra[] = <<<JS
+    <script>
+    $(document).on('ready', function() {
+        $('#reply-add-attachment').on('click', function(e) {
+            e.preventDefault();
+
+            var newInputFile = $('<input>', {
+                type: 'file',
+                name: 'user_upload[]'
+            });
+
+            $('[name="user_upload[]"]').parent().append(newInputFile);
+        });
+    });
+    </script>
+JS;
+
+if ($origin == 'learnpath') {
+    Display::display_reduced_header();
+} else {
+    Display :: display_header(null);
+}
+
+handle_forum_and_forumcategories();
+
+// Action links
+echo '<div class="actions">';
+echo '<span style="float:right;">'.search_link().'</span>';
+echo '<a href="viewforum.php?forum='.Security::remove_XSS($_GET['forum']).'&'.api_get_cidreq().'">'.
```

Comment:
if $_GET['forum'] is an integer (I suppose it is), then you should use intval() instead of Security::remove_XSS(). It is faster and more secure.